codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordDecoder.java

```diff
@@ -98,6 +153,14 @@ protected DnsRecord decodeRecord(
             return new DefaultDnsPtrRecord(
                     name, dnsClass, timeToLive, decodeName0(in.duplicate().setIndex(offset, offset + length)));
         }
+
+        if (type == DnsRecordType.TXT) {
+            String text = in.retainedDuplicate().setIndex(offset, offset + length).toString(CharsetUtil.US_ASCII);
```

Comment:
- TXT record cannot be longer than 255 bytes.
- Please make sure to check the maximum/minimum lengths of all record types you added.

Reply:
I'm not sure why the TXT record cannot be longer than 255, because [RFC 1035](https://tools.ietf.org/html/rfc1035#section-3.3.14) and [RFC 1464](https://tools.ietf.org/html/rfc1464) haven't limits, it seems base on the DNS server implementations, even the Wiki give a suggestion

> Value: This can be free form text data of any type. Each word is treated as a separate string unless one or more strings are enclosed in quotes. The maximum length of a string is 255 bytes, but any number of strings can be present in each record, subject to the standard DNS RDATA size limit of 65535 bytes.[8]